File: assignment1/q6.cpp
// clang++ -std=c++17 -Wall -Werror -Wextra -Wpedantic -g3 -o linelength linelength.cpp
// cd r28chang/SYDE-121/assignment01

#include <iostream>
#include <string>

auto main() -> int
{
    std::string curr;
    bool a = false; // let a be the first pair value
    bool b = false; // let b be the second pair value
    int lines = 0;  // line counter for file reader
    int pair = 0;   // pair counter for output

    while (getline(std::cin, curr))
    {
        lines++;

        // assinging proper boolean based on input
        if (lines % 2 == 1)
        {
            if (curr == "" || curr == "\n")
            {
                a = false;
            }
            else
            {
                a = true;
            }
        }

        else if (lines % 2 == 0)
        {
            if (curr == "" || curr == "\n")
            {
                b = false;
            }
            else
            {
                b = true;
            }
        }

        // output only for pairs
        if (lines % 2 == 0)
        {
            pair++;                                        // icnrement the pair number
            std::string pairString = std::to_string(pair); // convert to string

            // header
            std::cout << "Pair number: " + pairString << std::endl;
            std::cout << "----------------------------" << std::endl;

            // first case of De Morgan's Law
            std::cout << "not (a and b)" << std::endl;
            if (!(a && b))
            {
                std::cout << "true\n"
                          << std::endl;
            }
            else
            {
                std::cout << "false\n"
                          << std::endl;
            }

            // second case of De Morgan's Law
            std::cout << "not(a) or not(b)" << std::endl;
            if (!a || !b)
            {
                std::cout << "true\n"
                          << std::endl;
            }
            else
            {
                std::cout << "false\n"
                          << std::endl;
            }

            // third  case of De Morgan's Law
            std::cout << "not (a or b)" << std::endl;
            if (!(a || b))
            {
                std::cout << "true\n"
                          << std::endl;
            }
            else
            {
                std::cout << "false\n"
                          << std::endl;
            }

            // fourth case of De Morgan's Law
            std::cout << "not(a) and not(b)" << std::endl;
            if (!a && !b)
            {
                std::cout << "true\n"
                          << std::endl;
            }
            else
            {
                std::cout << "false\n"
                          << std::endl;
            }
        }
    }
    return 0;
}

File: assignment1/q6.py
import sys 

a = False  # let a be the first pair value 
b = False  # let b be the second pair value 
lines = 0 # line counter for file reader 
pair = 0 # pair counter for output 

# file reader 
for curr in sys.stdin:

    lines += 1

    # assign proper boolean based on input
    if (lines % 2 == 1): #odd number lines represent the a (first pair) value 
        # check for empty line 
        if (curr == "" or curr == "\n"):
            a = False
        else:
            a = True
    elif (lines % 2 == 0):  #even number lines represent the b (second pair) value 
        # check for empty line 
        if (curr == "" or curr == "\n"):
            b = False
        else:
            b = True

    # output (only occurs for each pair)
    if (lines % 2 == 0):
        pair += 1 #increment the pair number 

        print ("Pair number: " + str(pair))
        print ("-----------------------------")
        
        # first case of De Morgan's Law 
        print("not (a and b)")
        if (not( a and b)):
            print ("true\n")
        else: 
            print ("false\n")

        # second case of De Morgan's Law 
        print("not(a) or not(b))")
        if (not(a) or not(b)):
            print ("true\n")
        else: 
            print ("false\n")

        # third case of De Morgan's Law 
        print("not (a or b)")
        if (not(a or b)):
            print("true\n")
        else: 
            print ("false\n")

        # fourth case of De Morgan's Law 
        print("not(a) and not(b))")
        if (not(a) and not(b)):
            print ("true\n")
        else:
            print ("false\n")